api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -126,55 +126,139 @@ data LogRecordArguments body = LogRecordArguments
   , observedTimestamp :: Maybe Timestamp
   , context :: Maybe Context
   , severityText :: Maybe Text
-  , severityNumber :: Maybe Int
+  , severityNumber :: Maybe SeverityNumber
   , body :: body
   , attributes :: H.HashMap Text AnyValue
   }
 
 
-data SeverityNumber = SeverityNumber {shortName :: Maybe Text, severityInt :: !Int64} deriving (Read, Show)
+emptyLogRecordArguments :: body -> LogRecordArguments body
+emptyLogRecordArguments body =
+  LogRecordArguments
+    { timestamp = Nothing
+    , observedTimestamp = Nothing
+    , context = Nothing
+    , severityText = Nothing
+    , severityNumber = Nothing
+    , body = body
+    , attributes = H.empty
+    }
+
+
+data SeverityNumber
```

Comment:
An `Enum` is used so that setting `severityNumber` in `LogRecordArguments` does not need an `Int` with little semantic information. It is much easier to decide what number to use now because they are well labeled.